### PR TITLE
🔨 migrate charts from typeorm to knex

### PR DESCRIPTION
**Pull request by danyx23**
Migrates database access to the charts table and the chart_revisions table to knex

### CODE REVIEW

**Comment by danyx23** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3293?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3293** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3293?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3292** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3292?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3271** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3271?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3259** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3259?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3258** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3258?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @danyx23 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .vscode/launch.json [hunk 1]
```diff
@@ -104,6 +104,7 @@
             "program": "${workspaceFolder}/itsJustJavascript/adminSiteServer/app.js",
             "request": "launch",
             "type": "node",
+            "runtimeExecutable": "/mnt/wslg/runtime-dir/fnm_multishells/5693_1709636182174/bin/node"
```
Comment (marcelgerber) [inline .vscode/launch.json:107]: Don't think this should go here.

**Review by marcelgerber — COMMENTED** [review]
(partial review only, I didn't look at the heavy hitters yet)

File: db/db.ts [hunk 2]
```diff
@@ -156,6 +156,12 @@ export const knexRawFirst = async <TRow = unknown>(
     return results[0]
 }
 
+export const knexRawInsert = async (
```
[the remark this answers is not in the document]
Comment (danyx23) [reply db/db.ts:159]: Yeah good idea, in this case we can.

I'll make knexRawFirst take a RO transaction as this function is really for reading a single item I'd say

**Comment by danyx23** [comment]
### Merge activity

* **Mar 26, 10:04 AM EDT**: @danyx23 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/3293).

**Pull request merged**